src/subscriptions.c

```diff
@@ -952,6 +954,24 @@ subscription_done(void)
  * *************************************************************************/
 
 /**
+ * Update incoming byte count
+ */
+void subscription_add_bytes_in(th_subscription_t *s, size_t in)
+{
+  atomic_add(&s->ths_bytes_in, in);
+  atomic_add_u64(&s->ths_total_bytes_in, in);
```

Comment:
Unfortunately, this may cause a little synchronization issue (no lock for this update operation).

I would prefer to update only ths_total_bytes_in and ths_bytes_in should be the last read value, so the atomic_exchange calls should appear like:

<pre>
prev = s->ths_bytes_in;
s->ths_bytes_in_prev = atomic_exchange(&s->ths_bytes_in, s->ths_total_bytes_in) - prev;
</pre>